public/app/features/dashboard/state/actions.ts

```diff
@@ -120,7 +120,16 @@ export function initDashboardPanel(panel: PanelModel): ThunkResult<void> {
     }
 
     if (!panel.plugin) {
-      panel.pluginLoaded(plugin);
+      try {
+        panel.pluginLoaded(plugin);
+      } catch (err) {
+        if (err.changePanelType) {
+          console.log('Plugin loading requires chaning panel type:', err.changePanelType);
+          await dispatch(changePanelPlugin(panel, err.changePanelType));
+          return; // dispatch will send the required events
+        }
+        console.log('Error loading panel', err, panel);
```

Comment:
Maybe we should dispatch an error action here and return?

Reply:
I think we can handle this cleaner in a DashboardMigrartion, not sure there will be a general use case for this for other (external plugins), maybe we can find a new type of hook/info in plugin.json if that is the case to solve this more generally for external plugins (but do not think we have that yet).

Reply:
👍 I have never seen the DashboardMigrator!